features/frameworks/mbed-coap/source/sn_coap_builder.c

```diff
@@ -352,7 +349,12 @@ uint16_t sn_coap_builder_calc_needed_packet_data_size_2(const sn_coap_hdr_s *src
             returned_byte_count += src_coap_msg_ptr->payload_len;
         }
 #else
-        returned_byte_count += src_coap_msg_ptr->payload_len;
+        if ( sn_coap_builder_check_uint16_overflow(returned_byte_count,src_coap_msg_ptr->payload_len)){
```

Comment:
This is kind of the worst of all worlds for ARM code generation - you're using a 16-bit variable, which means every single addition in the routine has to have code inserted to cause 16-bit wrap, and then you're having to check manually whether it's happened. (But only here? You've missed payload addition above for blockwise, and aren't doing it for any options).

Assuming we're optimising for ARM rather than MSP430 now, and don't want to change any APIs, I'd suggest switching to use `size_t` internally, and do a simpler range check on the return as you cast down to `uint16_t`. That should make the code smaller, by getting rid of all the intermediate 16-bit wraps.

Reply:
I originally added check for every where byte count was increased, on those case it just could not overflow. size_t would be better solution.